### DISCUSSION TITLE
How do I extend an existing project using ALR

### DISCUSSION THREAD

**Discussion by Joebeazelman** [alire-project/alire#1032] [Q&A]
I am working on a small Ada program which imports another Ada application as a library. I've downloaded the Application's repository as a submodule and modified my main GPR file to:

```
with "config/hal_builder_config.gpr";

project Hal_Builder extends "svd2ada/svd2ada.gpr" is
end Hal_Builder;
```

When I build the project, Alire mysteriously added the following lines in my configuration file which wouldn't compile:

```
--  Configuration for hal_builder generated by Alire
with "utilada_base.gpr";
with "utilada_conf.gpr";
with "utilada_core.gpr";
with "utilada_sys.gpr";
abstract project hal_builder_Config is
   Crate_Version := "0.0.0";

end hal_builder_Config;

```
I reset my repository and added svd2ada as an ALR crate using the `with` clause, but it complained: 

```
   ✓ svd2ada * (add)
                                                              
Changes to dependency solution:

   New solution is incomplete.
   +⚠ svd2ada * (new,missing)
```

I went ahead and added it and it gave me an error:

`hal_builder_config.gpr:2:06: unknown project file: "utilada_base.gpr"`

What's the best way to setup a project which imports other dependencies while using ALR? Why is it including the phantom GPRs in my config file?

**Comment by Fabien-Chouteau** [comment]
Let's discuss in #1033 and not duplicate things.
